File: maps/src/main/java/it/localhostsoftware/maps/google/model/GoogleMarker.java
package it.localhostsoftware.maps.google.model;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import it.localhostsoftware.maps.model.BitmapDescriptor;
import it.localhostsoftware.maps.model.LatLng;
import it.localhostsoftware.maps.model.Marker;


public class GoogleMarker implements Marker {
    private final com.google.android.gms.maps.model.Marker marker;

    public GoogleMarker(com.google.android.gms.maps.model.Marker marker) {
        this.marker = marker;
    }

    @Override
    public void remove() {
        marker.remove();
    }

    @Override
    public String getId() {
        return marker.getId();
    }

    @Override
    public void setPosition(@NonNull LatLng latLng) {
        marker.setPosition(((GoogleLatLng) latLng).getGoogleLatLng());
    }

    @Override
    public LatLng getPosition() {
        return new GoogleLatLng(marker.getPosition());
    }

    @Override
    public void setZIndex(float v) {
        marker.setZIndex(v);
    }

    @Override
    public float getZIndex() {
        return marker.getZIndex();
    }

    @Override
    public void setIcon(@Nullable BitmapDescriptor bitmapDescriptor) {
        marker.setIcon(((GoogleBitmapDescriptor) bitmapDescriptor).getBitmapDescriptor());
    }

    @Override
    public void setAnchor(float v, float v1) {
        marker.setAnchor(v, v1);
    }

    @Override
    public void setInfoWindowAnchor(float v, float v1) {
        marker.setInfoWindowAnchor(v, v1);
    }

    @Override
    public void setTitle(@Nullable String s) {
        marker.setTitle(s);
    }

    @Override
    public String getTitle() {
        return marker.getTitle();
    }

    @Override
    public void setSnippet(@Nullable String s) {
        marker.setSnippet(s);
    }

    @Override
    public String getSnippet() {
        return marker.getSnippet();
    }

    @Override
    public void setDraggable(boolean b) {
        marker.setDraggable(b);
    }

    @Override
    public boolean isDraggable() {
        return marker.isDraggable();
    }

    @Override
    public void showInfoWindow() {
        marker.showInfoWindow();
    }

    @Override
    public void hideInfoWindow() {
        marker.hideInfoWindow();
    }

    @Override
    public boolean isInfoWindowShown() {
        return marker.isInfoWindowShown();
    }

    @Override
    public void setVisible(boolean b) {
        marker.setVisible(b);
    }

    @Override
    public boolean isVisible() {
        return marker.isVisible();
    }

    @Override
    public void setFlat(boolean b) {
        marker.setFlat(b);
    }

    @Override
    public boolean isFlat() {
        return marker.isFlat();
    }

    @Override
    public void setRotation(float v) {
        marker.setRotation(v);
    }

    @Override
    public float getRotation() {
        return marker.getRotation();
    }

    @Override
    public void setAlpha(float v) {
        marker.setAlpha(v);
    }

    @Override
    public float getAlpha() {
        return marker.getAlpha();
    }

    @Override
    public void setTag(@Nullable Object o) {
        marker.setTag(o);
    }

    @Nullable
    @Override
    public Object getTag() {
        return marker.getTag();
    }
}

File: maps/src/main/java/it/localhostsoftware/maps/google/model/GoogleMarker.kt
package it.localhostsoftware.maps.google.model

import com.google.android.gms.maps.model.Marker

class GoogleMarker(marker: Marker) : it.localhostsoftware.maps.model.Marker<Marker, GoogleLatLng, GoogleBitmapDescriptor>(marker) {
    override fun remove() {
        m.remove()
    }

    override val id: String
        get() = m.id
    override var position: GoogleLatLng
        get() = GoogleLatLng(m.position)
        set(value) {
            m.position = value.ll
        }
    override var zIndex: Float
        get() = m.zIndex
        set(v) {
            m.zIndex = v
        }

    override fun setIcon(var1: GoogleBitmapDescriptor?) {
        m.setIcon(var1?.bitmapDescriptor)
    }

    override fun setAnchor(var1: Float, var2: Float) {
        m.setAnchor(var1, var2)
    }

    override fun setInfoWindowAnchor(var1: Float, var2: Float) {
        m.setInfoWindowAnchor(var1, var2)
    }

    override var title: String?
        get() = m.title
        set(s) {
            m.title = s
        }
    override var snippet: String?
        get() = m.snippet
        set(s) {
            m.snippet = s
        }
    override var isDraggable: Boolean
        get() = m.isDraggable
        set(b) {
            m.isDraggable = b
        }

    override fun showInfoWindow() {
        m.showInfoWindow()
    }

    override fun hideInfoWindow() {
        m.hideInfoWindow()
    }

    override val isInfoWindowShown: Boolean
        get() = m.isInfoWindowShown
    override var isVisible: Boolean
        get() = m.isVisible
        set(b) {
            m.isVisible = b
        }
    override var isFlat: Boolean
        get() = m.isFlat
        set(b) {
            m.isFlat = b
        }
    override var rotation: Float
        get() = m.rotation
        set(v) {
            m.rotation = v
        }
    override var alpha: Float
        get() = m.alpha
        set(v) {
            m.alpha = v
        }
    override var tag: Any?
        get() = m.tag
        set(o) {
            m.tag = o
        }
}